staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go

```diff
@@ -112,7 +112,7 @@ func calculateCondition(in *apiextensions.CustomResourceDefinition) *apiextensio
 			return cond
 		}
 
-		pth := field.NewPath("spec", "version").Key(v.Name).Child("schema", "openAPIV3Schema")
```

Comment:
Is this an associative list? if so I think keyed by name is at least a somewhat acceptable way to index it. (version->versions seems correct though.)

But how to avoid fighting with a prior apiserver during an upgrade?

I'm not a fan of our current mechanism at all, it seems super brittle. Changes like this one shouldn't be so fraught with danger :/

Reply:
field path is indended to output the jsonpath to the field, so this should be an index

> But how to avoid fighting with a prior apiserver during an upgrade?

we already protected against that, I just wanted a sanity check the mechanism was working

Reply:
> field path is intended to output the jsonpath to the field, so this should be an index

Sounds like something that could (and therefore should) be tested...